### PR TITLE
feat: [PLG-3066]: Add secondary entitlement

### PR DESCRIPTION
**Pull request by Xin-Harness**
## Describe your changes
Add secondary entitlement in our system to support primary to secondary entitlement mapping
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56229)

### CODE REVIEW

**Comment by Xin-Harness** [comment]
trigger gitleaks

**Review by kapil-harness — APPROVED** [review]

**Review by neelam-harness — CHANGES_REQUESTED** [review]

File: 970-ng-commons/src/main/java/io/harness/entities/SecondaryEntitlement.java [hunk 1]
```diff
@@ -11,5 +11,21 @@
 
 @OwnedBy(HarnessTeam.GTM)
 public enum SecondaryEntitlement {
-  WORK_LOADS,
-}
+  NUMBER_OF_SERVICES("numberOfServices"), // CD service type license, SRM, CHAOS
```
[the remark this answers is not in the document]
Comment (Xin-Harness) [reply 970-ng-commons/src/main/java/io/harness/entities/SecondaryEntitlement.java:14]: The order of this is based on the mapping table that was listed in the wiki, it is the order GA order of each modules. I personally think using that order would make it easier to track. https://harness.atlassian.net/wiki/spaces/GTM/pages/21563704194/Tech+Spec+Default+developer+to+Secondary+Entitlement+Mapping+information

[the remark this answers is not in the document]
Comment (neelam-harness) [reply 970-ng-commons/src/main/java/io/harness/entities/SecondaryEntitlement.java:14]: I don't agree with that. The order in the code doesn't need to be the same as in the wiki. It needs to be more readable and general guideline for declaring the variables is to put them in alphabetically sorted order  for better readability otheriwse with the ever-growing list it becomes difficult to maintain it in a haphazard order.

File: 970-ng-commons/src/main/java/io/harness/entities/SecondaryEntitlement.java [hunk 2]
```diff
@@ -11,5 +11,28 @@
...
-}
+  NUMBER_OF_AGENTS("numberOfAgents"), // CET
+
+  NUMBER_OF_COMMITTERS("numberOfCommitters"), // CI
+
+  NUMBER_OF_CONTRIBUTORS("numberOfContributors"), // SEI
+
+  NUMBER_OF_CLIENT_MAUS("numberOfClientMAUs"), // FF
+
+  NUMBER_OF_EXECUTION_APPLIES("numberOfExecutionApplies"), // IACM
```
Comment (neelam-harness) [inline 970-ng-commons/src/main/java/io/harness/entities/SecondaryEntitlement.java:22]: nit: this one needs to be after number_of_developers

**Review by kapil-harness — DISMISSED** [review]

**Comment by Xin-Harness** [comment]
trigger uts

**Comment by Xin-Harness** [comment]
trigger compile

**Comment by Xin-Harness** [comment]
trigger compile

**Comment by ivan-levelops** [comment]
Merging now due to PR Checks failing for non related causes

**Pull request merged**